diff --git a/src/minecraft/net/minecraft/src/BlockLadder.java b/src/minecraft/net/minecraft/src/BlockLadder.java
index 4ac222a..ec1a2ee 100644
--- a/src/minecraft/net/minecraft/src/BlockLadder.java
+++ b/src/minecraft/net/minecraft/src/BlockLadder.java
@@ -175,0 +176,15 @@ public class BlockLadder extends Block
+
+    public int GetItemIDDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return Item.stick.itemID;
+    }
+
+    public int GetItemCountDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return 2;
+    }
+
+    public int GetItemDamageDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return 0;
+    }
