diff --git a/src/minecraft/net/minecraft/src/ComponentMineshaftCorridor.java b/src/minecraft/net/minecraft/src/ComponentMineshaftCorridor.java
index 3451ae6..3f90aa3 100644
--- a/src/minecraft/net/minecraft/src/ComponentMineshaftCorridor.java
+++ b/src/minecraft/net/minecraft/src/ComponentMineshaftCorridor.java
@@ -317 +317 @@ public boolean addComponentParts(World par1World, Random par2Random, StructureBo
-                        this.randomlyPlaceBlock(par1World, par3StructureBoundingBox, par2Random, 0.7F, 1, 0, var5, Block.rail.blockID, this.getMetadataWithOffset(Block.rail.blockID, 0));
+                        this.randomlyPlaceBlock(par1World, par3StructureBoundingBox, par2Random, 0.7F, 1, 0, var5, GPEBTWTweak.gpeBlockRustedRail.blockID, this.coordBaseMode & 1);
