diff --git a/src/minecraft/net/minecraft/src/EntityItem.java b/src/minecraft/net/minecraft/src/EntityItem.java
index 4c55875..ec95fce 100644
--- a/src/minecraft/net/minecraft/src/EntityItem.java
+++ b/src/minecraft/net/minecraft/src/EntityItem.java
@@ -125 +125 @@ public class EntityItem extends Entity
-        if (!this.worldObj.isRemote && this.age >= 6000)
+        if (!this.worldObj.isRemote && this.age >= 20*60*30)
