diff --git a/src/minecraft/net/minecraft/src/EntitySkeleton.java b/src/minecraft/net/minecraft/src/EntitySkeleton.java
index 310f2c7..89490ec 100644
--- a/src/minecraft/net/minecraft/src/EntitySkeleton.java
+++ b/src/minecraft/net/minecraft/src/EntitySkeleton.java
@@ -141 +141 @@ public void onLivingUpdate()
-            if (var1 > 0.5F && this.rand.nextFloat() * 30.0F < (var1 - 0.4F) * 2.0F && this.worldObj.canBlockSeeTheSky(MathHelper.floor_double(this.posX), MathHelper.floor_double(this.posY), MathHelper.floor_double(this.posZ)))
+            if (var1 > 0.5F && this.rand.nextFloat() * 30.0F < (var1 - 0.4F) * 2.0F && this.worldObj.canBlockSeeTheSky(MathHelper.floor_double(this.posX), MathHelper.floor_double(this.posY) + 1, MathHelper.floor_double(this.posZ)))
@@ -232,0 +233,10 @@ protected void dropFewItems(boolean par1, int par2)
+
+        if (this.getSkeletonType() == 1)
+        {
+            var3 = this.rand.nextInt(3 + par2) - 1;
+
+            for (var4 = 0; var4 < var3; ++var4)
+            {
+                this.dropItem(GPEBTWTweak.gpeItemAsh.itemID, 1);
+            }
+        }
