diff --git a/src/minecraft/net/minecraft/src/EntityCreeper.java b/src/minecraft/net/minecraft/src/EntityCreeper.java
index f5f68d7..ecfb838 100644
--- a/src/minecraft/net/minecraft/src/EntityCreeper.java
+++ b/src/minecraft/net/minecraft/src/EntityCreeper.java
@@ -280,0 +281,2 @@ public class EntityCreeper extends EntityMob
+            this.attackEntityFrom(DamageSource.causePlayerDamage(var1), 2);
+
