diff --git a/src/minecraft/net/minecraft/src/EntityChicken.java b/src/minecraft/net/minecraft/src/EntityChicken.java
index e306837..45c7131 100644
--- a/src/minecraft/net/minecraft/src/EntityChicken.java
+++ b/src/minecraft/net/minecraft/src/EntityChicken.java
@@ -141 +141 @@ public class EntityChicken extends EntityAnimal
-        int var3 = this.rand.nextInt(3) + this.rand.nextInt(1 + par2);
+        int var3 = 1 + this.rand.nextInt(3) + this.rand.nextInt(1 + par2);
