diff --git a/src/minecraft/net/minecraft/src/EntityLiving.java b/src/minecraft/net/minecraft/src/EntityLiving.java
index a2a98c4..df2f23d 100644
--- a/src/minecraft/net/minecraft/src/EntityLiving.java
+++ b/src/minecraft/net/minecraft/src/EntityLiving.java
@@ -2871,0 +2872,5 @@ public abstract class EntityLiving extends Entity
+    public void setPersistent()
+    {
+        this.persistenceRequired = true;
+    }
+
