diff --git a/src/minecraft/net/minecraft/src/EntityRenderer.java b/src/minecraft/net/minecraft/src/EntityRenderer.java
index 1d66344..6cd1f76 100644
--- a/src/minecraft/net/minecraft/src/EntityRenderer.java
+++ b/src/minecraft/net/minecraft/src/EntityRenderer.java
@@ -1142,0 +1143 @@ public void renderWorld(float par1, long par2)
+                    GPEBTWTweakProxyClient.drawMicroblockTarget(par1);
