diff --git a/src/minecraft/net/minecraft/src/EntityItem.java b/src/minecraft/net/minecraft/src/EntityItem.java
index 4c55875..f69e385 100644
--- a/src/minecraft/net/minecraft/src/EntityItem.java
+++ b/src/minecraft/net/minecraft/src/EntityItem.java
@@ -125 +125 @@ public void onUpdate()
-        if (!this.worldObj.isRemote && this.age >= 6000)
+        if (!this.worldObj.isRemote && this.age >= 20*60*30)
@@ -300,0 +301,5 @@ public void onCollideWithPlayer(EntityPlayer par1EntityPlayer)
+            if (var3 <= 0)
+            {
+              this.setDead();
+              return;
+            }
@@ -323,0 +329,5 @@ public void onCollideWithPlayer(EntityPlayer par1EntityPlayer)
+                if (var2.itemID == Block.enchantmentTable.blockID)
+                {
+                    par1EntityPlayer.triggerAchievement(AchievementList.enchantments);
+                }
+
@@ -456,0 +467 @@ protected void doBlockCollisions()
+                            GPEBTWTweak.onEntityCollidedWithBlock(this.worldObj, var7, var8, var9, this);
