diff --git a/src/minecraft/net/minecraft/src/EntityPlayer.java b/src/minecraft/net/minecraft/src/EntityPlayer.java
index dc30dad..42fa487 100644
--- a/src/minecraft/net/minecraft/src/EntityPlayer.java
+++ b/src/minecraft/net/minecraft/src/EntityPlayer.java
@@ -258 +258 @@ public abstract class EntityPlayer extends EntityLiving implements ICommandSende
-                    this.wakeUpPlayer(false, true, true);
+                    //this.wakeUpPlayer(false, true, true);
@@ -1333 +1333 @@ public abstract class EntityPlayer extends EntityLiving implements ICommandSende
-    public EnumStatus sleepInBedAt(int par1, int par2, int par3)
+    public EnumStatus sleepInBedAt(int var1, int var2, int var3)
@@ -1335 +1335,55 @@ public abstract class EntityPlayer extends EntityLiving implements ICommandSende
-        return EnumStatus.OTHER_PROBLEM;
+        if (!this.worldObj.isRemote)
+        {
+            if (this.isPlayerSleeping() || !this.isEntityAlive())
+            {
+                return EnumStatus.OTHER_PROBLEM;
+            }
+
+            if (Math.abs(this.posX - (double)var1) > 3.0D || Math.abs(this.posY - (double)var2) > 2.0D || Math.abs(this.posZ - (double)var3) > 3.0D)
+            {
+                return EnumStatus.TOO_FAR_AWAY;
+            }
+        }
+
+        this.setSize(0.2F, 0.2F);
+        this.yOffset = 0.2F;
+
+        if (this.worldObj.blockExists(var1, var2, var3))
+        {
+            int var9 = this.worldObj.getBlockMetadata(var1, var2, var3);
+            int var5 = BlockBed.getDirection(var9);
+            float var10 = 0.5F;
+            float var7 = 0.5F;
+
+            switch (var5)
+            {
+                case 0:
+                    var7 = 0.9F;
+                    break;
+
+                case 1:
+                    var10 = 0.1F;
+                    break;
+
+                case 2:
+                    var7 = 0.1F;
+                    break;
+
+                case 3:
+                    var10 = 0.9F;
+            }
+
+            this.func_71013_b(var5);
+            this.setPosition((double)((float)var1 + var10), (double)((float)var2 + 0.9375F), (double)((float)var3 + var7));
+        }
+        else
+        {
+            this.setPosition((double)((float)var1 + 0.5F), (double)((float)var2 + 0.9375F), (double)((float)var3 + 0.5F));
+        }
+
+        this.sleeping = true;
+        this.sleepTimer = 0;
+        this.playerLocation = new ChunkCoordinates(var1, var2, var3);
+        this.motionX = this.motionZ = this.motionY = 0.0D;
+
+        return EnumStatus.OK;
@@ -1403 +1457 @@ public abstract class EntityPlayer extends EntityLiving implements ICommandSende
-            this.setSpawnChunk(this.playerLocation, false);
+            //this.setSpawnChunk(this.playerLocation, false);
@@ -1484 +1538 @@ public abstract class EntityPlayer extends EntityLiving implements ICommandSende
-        return this.sleeping && this.sleepTimer >= 100;
+        return false;
@@ -2249 +2303 @@ public abstract class EntityPlayer extends EntityLiving implements ICommandSende
-        if (var1 >= 4)
+        if (var1 >= 4 || getFlag(4)) // isEating()
@@ -2406 +2460,6 @@ public abstract class EntityPlayer extends EntityLiving implements ICommandSende
-        return this.GetHealthAndExhaustionModifierWithSightlessModifier();
+        float f = this.GetHealthAndExhaustionModifierWithSightlessModifier();
+        if (this.inventory.armorInventory[0] != null)
+        {
+          f *= 1.1F + 0.1F * EnchantmentHelper.getEnchantmentLevel(GPEBTWTweak.gpeEnchantmentHaste, this.inventory.armorInventory[0]);
+        }
+        return f * (1.0F - ((float)this.GetWornArmorWeight() / 220.0F));
