diff --git a/src/minecraft/net/minecraft/src/EntityZombie.java b/src/minecraft/net/minecraft/src/EntityZombie.java
index 3466d92..07416af 100644
--- a/src/minecraft/net/minecraft/src/EntityZombie.java
+++ b/src/minecraft/net/minecraft/src/EntityZombie.java
@@ -142 +142 @@ public class EntityZombie extends EntityMob
-            if (var1 > 0.5F && this.rand.nextFloat() * 30.0F < (var1 - 0.4F) * 2.0F && this.worldObj.canBlockSeeTheSky(MathHelper.floor_double(this.posX), MathHelper.floor_double(this.posY), MathHelper.floor_double(this.posZ)))
+            if (var1 > 0.5F && this.rand.nextFloat() * 30.0F < (var1 - 0.4F) * 2.0F && this.worldObj.canBlockSeeTheSky(MathHelper.floor_double(this.posX), MathHelper.floor_double(this.posY) + 1, MathHelper.floor_double(this.posZ)))
