diff --git a/src/minecraft/net/minecraft/src/EntityBlaze.java b/src/minecraft/net/minecraft/src/EntityBlaze.java
index 0ecaafd..3712712 100644
--- a/src/minecraft/net/minecraft/src/EntityBlaze.java
+++ b/src/minecraft/net/minecraft/src/EntityBlaze.java
@@ -258,0 +259,5 @@ public void CheckForScrollDrop()
+
+    protected void dropHead()
+    {
+        this.entityDropItem(new ItemStack(Item.skull.itemID, 1, 8), 0.0F);
+    }
