diff --git a/src/minecraft/net/minecraft/src/FCBetterThanWolves.java b/src/minecraft/net/minecraft/src/FCBetterThanWolves.java
index c06e4fe..8368cf1 100644
--- a/src/minecraft/net/minecraft/src/FCBetterThanWolves.java
+++ b/src/minecraft/net/minecraft/src/FCBetterThanWolves.java
@@ -813 +813 @@ public class FCBetterThanWolves extends FCAddOn
-        this.AddItemName(var1, Item.stick, "Shaft");
+        this.AddItemName(var1, Item.stick, "Rod");
@@ -951,0 +952,4 @@ public class FCBetterThanWolves extends FCAddOn
+        var12 = fcBlockDirtSlab.getUnlocalizedName() + ".gravel" + ".name";
+        var1.GetTranslateTable().put(var12, "Gravel Slab");
+        var12 = fcBlockDirtSlab.getUnlocalizedName() + ".sand" + ".name";
+        var1.GetTranslateTable().put(var12, "Sand Slab");
@@ -957,0 +962,4 @@ public class FCBetterThanWolves extends FCAddOn
+        var1.GetTranslateTable().put("item.skull.spider.name", "Spider head");
+        var1.GetTranslateTable().put("item.skull.enderman.name", "Enderman head");
+        var1.GetTranslateTable().put("item.skull.pigzombie.name", "Zombie Pigman head");
+        var1.GetTranslateTable().put("item.skull.fire.name", "Blaze head");
@@ -1368 +1376 @@ public class FCBetterThanWolves extends FCAddOn
-        fcPotash = (new Item(fcPotashID)).setUnlocalizedName("fcItemPotash");
+        fcPotash = (new GPEItemPotash(fcPotashID)).setUnlocalizedName("fcItemPotash");
