diff --git a/src/minecraft/net/minecraft/src/FCBetterThanWolves.java b/src/minecraft/net/minecraft/src/FCBetterThanWolves.java
index 62631d7..a69214b 100644
--- a/src/minecraft/net/minecraft/src/FCBetterThanWolves.java
+++ b/src/minecraft/net/minecraft/src/FCBetterThanWolves.java
@@ -17 +17 @@ public class FCBetterThanWolves extends FCAddOn
-    public static final String fcVersionString = "4.8911";
+    public static String fcVersionString = "4.8911";
@@ -242,0 +243 @@ public class FCBetterThanWolves extends FCAddOn
+    public static Item fcItemWool;
