diff --git a/src/minecraft/net/minecraft/src/FCBlockAxle.java b/src/minecraft/net/minecraft/src/FCBlockAxle.java
index 20f0b8c..183c36b 100644
--- a/src/minecraft/net/minecraft/src/FCBlockAxle.java
+++ b/src/minecraft/net/minecraft/src/FCBlockAxle.java
@@ -10,0 +11,4 @@ public class FCBlockAxle extends Block implements FCIBlock
+    public Icon m_IconSideOn2;
+    public Icon m_IconSideOn3;
+    public Icon m_IconSideOn4;
+    public Icon m_IconSideOn4f;
@@ -535 +539,5 @@ public class FCBlockAxle extends Block implements FCIBlock
-        this.m_IconSideOn = var1.registerIcon("fcBlockAxle_side_on");
+        this.m_IconSideOn = var1.registerIcon("fcBlockAxle_side_on1");
+        this.m_IconSideOn2 = var1.registerIcon("fcBlockAxle_side_on2");
+        this.m_IconSideOn3 = var1.registerIcon("fcBlockAxle_side_on3");
+        this.m_IconSideOn4 = var1.registerIcon("fcBlockAxle_side_on4");
+        this.m_IconSideOn4f = var1.registerIcon("fcBlockAxle_side_on4f");
@@ -557 +565,8 @@ public class FCBlockAxle extends Block implements FCIBlock
-                return this.GetAxleSideTextureForOnState(this.m_bIsPowerOnForCurrentRender);
+                if (!this.m_bIsPowerOnForCurrentRender) return this.m_IconSide;
+                switch (var5)
+                {
+                    case 3: return this.m_IconSideOn;
+                    case 5: return this.m_IconSideOn2;
+                    case 2: return this.m_IconSideOn3;
+                    case 4: return this.m_IconSideOn4f;
+                }
@@ -564 +579,8 @@ public class FCBlockAxle extends Block implements FCIBlock
-                return this.GetAxleSideTextureForOnState(this.m_bIsPowerOnForCurrentRender);
+                if (!this.m_bIsPowerOnForCurrentRender) return this.m_IconSide;
+                switch (var5)
+                {
+                    case 0: return this.m_IconSideOn;
+                    case 5: return this.m_IconSideOn2;
+                    case 1: return this.m_IconSideOn3;
+                    case 4: return this.m_IconSideOn4;
+                }
@@ -569 +591,8 @@ public class FCBlockAxle extends Block implements FCIBlock
-            return this.GetAxleSideTextureForOnState(this.m_bIsPowerOnForCurrentRender);
+            if (!this.m_bIsPowerOnForCurrentRender) return this.m_IconSide;
+            switch (var5)
+            {
+                case 0: return this.m_IconSideOn;
+                case 2: return this.m_IconSideOn2;
+                case 1: return this.m_IconSideOn3;
+                case 3: return this.m_IconSideOn4;
+            }
