diff --git a/src/minecraft/net/minecraft/src/FCBlockBlockDispenser.java b/src/minecraft/net/minecraft/src/FCBlockBlockDispenser.java
index 0f0a35e..7629a9d 100644
--- a/src/minecraft/net/minecraft/src/FCBlockBlockDispenser.java
+++ b/src/minecraft/net/minecraft/src/FCBlockBlockDispenser.java
@@ -759,0 +760,10 @@ else if (var17.itemID == Item.snowball.itemID)
+                                else if (var17.itemID == GPEBTWTweak.gpeItemLooseRock.itemID)
+                                {
+                                    var34 = FCUtilsInventory.CountItemsInInventory(var16, var17.itemID, 32767);
+                                    if (var34 >= 3)
+                                    {
+                                        FCUtilsInventory.ConsumeItemsInInventory(var16, var17.itemID, 32767, 3);
+                                        var33 = Block.cobblestone;
+                                    }
+                                    else var12 = true;
+                                }
