diff --git a/src/minecraft/net/minecraft/src/FCBlockHopper.java b/src/minecraft/net/minecraft/src/FCBlockHopper.java
index 1e8c5b4..65a6529 100644
--- a/src/minecraft/net/minecraft/src/FCBlockHopper.java
+++ b/src/minecraft/net/minecraft/src/FCBlockHopper.java
@@ -495,4 +495,3 @@ public class FCBlockHopper extends BlockContainer implements FCIBlockMechanical,
-                            ItemStack var17 = new ItemStack(Item.flint.itemID, var20, 0);
-                            EntityItem var18 = new EntityItem(var1, var5.posX, var5.posY, var5.posZ, var17);
-                            var18.delayBeforeCanPickup = 10;
-                            var1.spawnEntityInWorld(var18);
+                            int numflints = (int)(0.5f + var20 * 0.2f * var1.rand.nextFloat());
+                            handleNewGravelTop(var1, var5, GPEBTWTweak.gpeItemLooseRock.itemID, var20 - numflints);
+                            handleNewGravelTop(var1, var5, Item.flint.itemID, numflints);
@@ -538,0 +538,8 @@ public class FCBlockHopper extends BlockContainer implements FCIBlockMechanical,
+    private void handleNewGravelTop(World world, EntityItem ei, int id, int amount)
+    {
+        if (amount <= 0) return;
+        EntityItem ni = new EntityItem(world, ei.posX, ei.posY, ei.posZ, new ItemStack(id, amount, 0));
+        ni.delayBeforeCanPickup = 10;
+        world.spawnEntityInWorld(ni);
+    }
+
