diff --git a/src/minecraft/net/minecraft/src/FCBlockLog.java b/src/minecraft/net/minecraft/src/FCBlockLog.java
index aefdee9..bd374db 100644
--- a/src/minecraft/net/minecraft/src/FCBlockLog.java
+++ b/src/minecraft/net/minecraft/src/FCBlockLog.java
@@ -57,0 +58,15 @@ public class FCBlockLog extends BlockLog
+
+    public float getBlockHardness(World world, int x, int y, int z)
+    {
+        float multiplier = 1;
+        // If dirt is below vertical log, consider it a hard stump
+        int meta = world.getBlockMetadata(x, y, z);
+        if (world.getBlockId(x, y - 1, z) == 3 && (meta & 12) == 0)
+        {
+          multiplier = 10;
+        }
+        // Make jungle wood somewhat softer, pine a bit softer
+        if ((meta & 3) == 1) return 1.3F * multiplier;
+        if ((meta & 3) == 3) return 1.0F * multiplier;
+        return 1.5F * multiplier;
+    }
