diff --git a/src/minecraft/net/minecraft/src/FCBlockAxle.java b/src/minecraft/net/minecraft/src/FCBlockAxle.java
index 20f0b8c..9811412 100644
--- a/src/minecraft/net/minecraft/src/FCBlockAxle.java
+++ b/src/minecraft/net/minecraft/src/FCBlockAxle.java
@@ -10,0 +11 @@ public class FCBlockAxle extends Block implements FCIBlock
+    private Icon[] niceTexLookup;
@@ -535 +536,17 @@ public class FCBlockAxle extends Block implements FCIBlock
-        this.m_IconSideOn = var1.registerIcon("fcBlockAxle_side_on");
+        this.m_IconSideOn = var1.registerIcon("fcBlockAxle_side_on1");
+        Icon m_IconSideOn2 = var1.registerIcon("fcBlockAxle_side_on2");
+        Icon m_IconSideOn3 = var1.registerIcon("fcBlockAxle_side_on3");
+        Icon m_IconSideOn4 = var1.registerIcon("fcBlockAxle_side_on4");
+        Icon m_IconSideOn4f = var1.registerIcon("fcBlockAxle_side_on4f");
+        niceTexLookup = new Icon[] {
+            m_IconSideOn,   // 0     [*  [*
+            m_IconSideOn3,  // 1      *   *
+            m_IconSideOn2,  // 2          *
+            m_IconSideOn4,  // 3          *]
+            m_IconSideOn4,  // 4 [    *
+            m_IconSideOn2,  // 5      *]
+            m_IconSideOn3,  // 6  *
+            m_IconSideOn,   // 7  *
+            m_IconSideOn4f, // 8  *
+            m_IconSideOn2   // 9  *]
+        };
@@ -557 +574 @@ public class FCBlockAxle extends Block implements FCIBlock
-                return this.GetAxleSideTextureForOnState(this.m_bIsPowerOnForCurrentRender);
+                return getNiceSideTexture(4 + var5);
@@ -564 +581 @@ public class FCBlockAxle extends Block implements FCIBlock
-                return this.GetAxleSideTextureForOnState(this.m_bIsPowerOnForCurrentRender);
+                return getNiceSideTexture(var5);
@@ -569 +586 @@ public class FCBlockAxle extends Block implements FCIBlock
-            return this.GetAxleSideTextureForOnState(this.m_bIsPowerOnForCurrentRender);
+            return getNiceSideTexture(var5);
@@ -574,0 +592,6 @@ public class FCBlockAxle extends Block implements FCIBlock
+    public Icon getNiceSideTexture(int index)
+    {
+        if (!m_bIsPowerOnForCurrentRender) return m_IconSide;
+        return niceTexLookup[index];
+    }
+
