diff --git a/src/minecraft/net/minecraft/src/FCBlockPumpkin.java b/src/minecraft/net/minecraft/src/FCBlockPumpkin.java
index 61f1098..e082dc6 100644
--- a/src/minecraft/net/minecraft/src/FCBlockPumpkin.java
+++ b/src/minecraft/net/minecraft/src/FCBlockPumpkin.java
@@ -7,0 +8 @@ public class FCBlockPumpkin extends FCBlockGourd
+    private boolean isLantern;
@@ -11,0 +13 @@ public class FCBlockPumpkin extends FCBlockGourd
+        this.isLantern = var2;
@@ -128 +130 @@ public class FCBlockPumpkin extends FCBlockGourd
-        this.m_IconFront = var1.registerIcon("pumpkin_face");
+        this.m_IconFront = var1.registerIcon(this.isLantern ? "pumpkin_jack" : "pumpkin_face");
