diff --git a/src/minecraft/net/minecraft/src/FCBlockSaw.java b/src/minecraft/net/minecraft/src/FCBlockSaw.java
index 51f4f33..8aa4334 100644
--- a/src/minecraft/net/minecraft/src/FCBlockSaw.java
+++ b/src/minecraft/net/minecraft/src/FCBlockSaw.java
@@ -575,0 +576,5 @@ public class FCBlockSaw extends Block implements FCIBlock, FCIBlockMechanical, F
+            else if (var10 == Block.pressurePlateStone.blockID || var10 == Block.pressurePlatePlanks.blockID)
+            {
+                return true;
+            }
+            else var12 = GPEBTWTweak.instance.OnBlockSawed(var1, var2, var3, var4);
