diff --git a/src/minecraft/net/minecraft/src/FCBlockPlatform.java b/src/minecraft/net/minecraft/src/FCBlockPlatform.java
index 60c7aa0..2020cc7 100644
--- a/src/minecraft/net/minecraft/src/FCBlockPlatform.java
+++ b/src/minecraft/net/minecraft/src/FCBlockPlatform.java
@@ -341,0 +342,15 @@ public class FCBlockPlatform extends Block implements FCIBlock, FCIBlockSolidTop
+
+    public boolean OnBlockSawed(World world, int x, int y, int z)
+    {
+        int i;
+        for (i = 0; i < 3; ++i)
+        {
+            FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, FCBetterThanWolves.fcBlockWoodMouldingItemStubID, 0);
+        }
+        for (i = 0; i < 2; ++i)
+        {
+            FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, FCBetterThanWolves.fcWicker.itemID, 0);
+        }
+        world.setBlockToAir(x, y, z);
+        return true;
+    }
