diff --git a/src/minecraft/net/minecraft/src/FCItemChisel.java b/src/minecraft/net/minecraft/src/FCItemChisel.java
new file mode 100644
index 0000000..fbda3c3
--- /dev/null
+++ b/src/minecraft/net/minecraft/src/FCItemChisel.java
@@ -0,0 +1,10 @@
+package net.minecraft.src;
+
+// Dummy class for BTW 4.AA+
+public class FCItemChisel extends FCItemTool
+{
+  public FCItemChisel(int id, EnumToolMaterial mat, int uses)
+  {
+    super(id, 1, mat);
+  }
+}
