diff --git a/src/minecraft/net/minecraft/src/FCItemBlockTorchBurning.java b/src/minecraft/net/minecraft/src/FCItemBlockTorchBurning.java
new file mode 100644
index 0000000..ca5c184
--- /dev/null
+++ b/src/minecraft/net/minecraft/src/FCItemBlockTorchBurning.java
@@ -0,0 +1,10 @@
+package net.minecraft.src;
+
+// Dummy class for BTW 4.AA+
+public class FCItemBlockTorchBurning extends Item
+{
+  public FCItemBlockTorchBurning(int id)
+  {
+    super(id);
+  }
+}
