diff --git a/src/minecraft/net/minecraft/src/FCTileEntityCookingVessel.java b/src/minecraft/net/minecraft/src/FCTileEntityCookingVessel.java
index 9c70521..40900ab 100644
--- a/src/minecraft/net/minecraft/src/FCTileEntityCookingVessel.java
+++ b/src/minecraft/net/minecraft/src/FCTileEntityCookingVessel.java
@@ -7 +7 @@ public abstract class FCTileEntityCookingVessel extends TileEntity implements II
-    protected final int m_iInventorySize = 27;
+    protected final int m_iInventorySize = 9;
@@ -15 +15 @@ public abstract class FCTileEntityCookingVessel extends TileEntity implements II
-    protected ItemStack[] m_Contents = new ItemStack[27];
+    protected ItemStack[] m_Contents = new ItemStack[m_iInventorySize];
@@ -148 +148 @@ public abstract class FCTileEntityCookingVessel extends TileEntity implements II
-                        this.m_iCookCounter = 0;
+                        this.m_iCookCounter = Math.max(this.m_iCookCounter - 24, 0);
@@ -184 +184 @@ public abstract class FCTileEntityCookingVessel extends TileEntity implements II
-        return 27;
+        return m_iInventorySize;
@@ -191,0 +192 @@ public abstract class FCTileEntityCookingVessel extends TileEntity implements II
+        if (var1 >= m_iInventorySize) return null;
@@ -226,0 +228 @@ public abstract class FCTileEntityCookingVessel extends TileEntity implements II
+        if (var1 >= m_iInventorySize) return;
