diff --git a/src/minecraft/net/minecraft/src/FCTileEntityTurntable.java b/src/minecraft/net/minecraft/src/FCTileEntityTurntable.java
index cbbe356..90dc0be 100644
--- a/src/minecraft/net/minecraft/src/FCTileEntityTurntable.java
+++ b/src/minecraft/net/minecraft/src/FCTileEntityTurntable.java
@@ -7,0 +8 @@
+    public int visualSpin = 0;
@@ -60,0 +62,3 @@ public void updateEntity()
+                    // This generates sequence 0, 2, 3, 1, [0, 2, 3, 1]...
+                    this.visualSpin = ((this.visualSpin >> 1) + 2 - (this.visualSpin << 1)) & 3;
+                    this.worldObj.markBlockRangeForRenderUpdate(this.xCoord, this.yCoord, this.zCoord, this.xCoord, this.yCoord, this.zCoord);
@@ -542,2 +546,2 @@ private void RotateFurnace(int var1, int var2, int var3, boolean var4)
-        int var6 = FCUtilsMisc.RotateFacingAroundJ(var5, var4);
-        this.worldObj.setBlockMetadataWithNotify(var1, var2, var3, var6);
+        int var6 = FCUtilsMisc.RotateFacingAroundJ(var5 & 7, var4);
+        this.worldObj.setBlockMetadataWithNotify(var1, var2, var3, var6 | (var5 & 8));
