diff --git a/src/minecraft/net/minecraft/src/FCUtilsWorld.java b/src/minecraft/net/minecraft/src/FCUtilsWorld.java
index b1736df..4c23016 100644
--- a/src/minecraft/net/minecraft/src/FCUtilsWorld.java
+++ b/src/minecraft/net/minecraft/src/FCUtilsWorld.java
@@ -248,0 +249,4 @@ else if (!var0.isBlockNormalCube(var1, var2, var3) && !var0.isBlockNormalCube(va
+            if (var4 == Block.mycelium.blockID || var4 == Block.mushroomCapBrown.blockID || var4 == Block.mushroomCapRed.blockID)
+            {
+                return false;
+            }
@@ -319,0 +324,6 @@ public static boolean DoesBlockHaveSolidTopSurface(IBlockAccess var0, int var1,
+
+    // dummy method for BTW 4.AAAAAAABBB+
+    public static long GetOverworldTimeServerOnly()
+    {
+        return 0;
+    }
