diff --git a/src/minecraft/net/minecraft/src/GuiMainMenu.java b/src/minecraft/net/minecraft/src/GuiMainMenu.java
index 2eab63c..d81206b 100644
--- a/src/minecraft/net/minecraft/src/GuiMainMenu.java
+++ b/src/minecraft/net/minecraft/src/GuiMainMenu.java
@@ -540,0 +541,2 @@ public void drawScreen(int par1, int par2, float par3)
+        this.drawString(this.fontRenderer, "BTW v" + FCBetterThanWolves.fcVersionString, 2, this.height - 30, 0xFFFF00);
+        this.drawString(this.fontRenderer, "BTWTweak v" + GPEBTWTweak.tweakVersion, 2, this.height - 20, 0x55AAFF);
