diff --git a/src/minecraft/net/minecraft/client/Minecraft.java b/src/minecraft/net/minecraft/client/Minecraft.java
index 60d1a40..51fea38 100644
--- a/src/minecraft/net/minecraft/client/Minecraft.java
+++ b/src/minecraft/net/minecraft/client/Minecraft.java
@@ -54,0 +55,2 @@ import net.minecraft.src.GameWindowListener;
+import net.minecraft.src.GPEBTWTweak;
+import net.minecraft.src.GPEBTWTweakProxyClient;
@@ -471,0 +474,2 @@ public abstract class Minecraft implements Runnable, IPlayerUsage
+
+        GPEBTWTweak.readyForInput();
@@ -1622,0 +1627 @@ public abstract class Minecraft implements Runnable, IPlayerUsage
+                            GPEBTWTweak.onKeyPress(Keyboard.getEventKey());
@@ -2261,0 +2267 @@ public abstract class Minecraft implements Runnable, IPlayerUsage
+        GPEBTWTweakProxyClient.setAppIcon(var17);
