diff --git a/src/minecraft/net/minecraft/src/NetServerHandler.java b/src/minecraft/net/minecraft/src/NetServerHandler.java
index 1a2c49d..22e8d21 100644
--- a/src/minecraft/net/minecraft/src/NetServerHandler.java
+++ b/src/minecraft/net/minecraft/src/NetServerHandler.java
@@ -1007 +1007 @@ public class NetServerHandler extends NetHandler
-            this.playerEntity.playerNetServerHandler.sendPacketToPlayer(new Packet3Chat("\u00a74" + "This server requires all clients to have Better Than Wolves version " + "4.8911"));
+            this.playerEntity.playerNetServerHandler.sendPacketToPlayer(new Packet3Chat("\u00a74" + "This server requires all clients to have Better Than Wolves version " + FCBetterThanWolves.fcVersionString));
