diff --git a/src/minecraft/net/minecraft/src/PlayerControllerMP.java b/src/minecraft/net/minecraft/src/PlayerControllerMP.java
index bbc8e81..5f81d49 100644
--- a/src/minecraft/net/minecraft/src/PlayerControllerMP.java
+++ b/src/minecraft/net/minecraft/src/PlayerControllerMP.java
@@ -163,0 +164,4 @@ public void clickBlock(int par1, int par2, int par3, int par4)
+        if (this.currentGameType.isCreative() && this.mc.thePlayer.getHeldItem() != null && this.mc.thePlayer.getHeldItem().getItem() instanceof ItemSword)
+        {
+            return;
+        }
@@ -225,0 +230,4 @@ public void onPlayerDamageBlock(int par1, int par2, int par3, int par4)
+        if (this.currentGameType.isCreative() && this.mc.thePlayer.getHeldItem() != null && this.mc.thePlayer.getHeldItem().getItem() instanceof ItemSword)
+        {
+            return;
+        }
@@ -347 +355,2 @@ public boolean onPlayerRightClick(EntityPlayer par1EntityPlayer, World par2World
-            if (!var14.canPlaceItemBlockOnSide(par2World, par4, par5, par6, par7, par1EntityPlayer, par3ItemStack))
+            if (!GPEBTWTweak.canPlaceItemBlock(var14, par2World, par4, par5, par6, par7, par1EntityPlayer, par3ItemStack, var9, var10, var11))
+            //if (!var14.canPlaceItemBlockOnSide(par2World, par4, par5, par6, par7, par1EntityPlayer, par3ItemStack))
