diff --git a/src/minecraft/net/minecraft/src/ServerConfigurationManager.java b/src/minecraft/net/minecraft/src/ServerConfigurationManager.java
index caddc97..02ed356 100644
--- a/src/minecraft/net/minecraft/src/ServerConfigurationManager.java
+++ b/src/minecraft/net/minecraft/src/ServerConfigurationManager.java
@@ -1005 +1005 @@ private boolean AssignNewHardcoreSpawnLocation(World var1, EntityPlayerMP var2)
-            double var4 = var1.rand.nextDouble() * 2000.0D;
+            double var4 = var1.rand.nextDouble() * (double)GPEBTWTweak.hcSpawnRadius;
