diff --git a/src/minecraft/net/minecraft/src/TileEntityFurnace.java b/src/minecraft/net/minecraft/src/TileEntityFurnace.java
index f28d825..b3a08e7 100644
--- a/src/minecraft/net/minecraft/src/TileEntityFurnace.java
+++ b/src/minecraft/net/minecraft/src/TileEntityFurnace.java
@@ -20,0 +21 @@ public class TileEntityFurnace extends TileEntity implements ISidedInventory
+    public int currentItemID = 0;
@@ -243,0 +245 @@ public class TileEntityFurnace extends TileEntity implements ISidedInventory
+        boolean shouldUpdate = false;
@@ -273 +275,13 @@ public class TileEntityFurnace extends TileEntity implements ISidedInventory
-            if (this.isBurning() && this.canSmelt())
+            if (this.currentItemID == 0 && this.furnaceItemStacks[0] != null)
+            {
+                shouldUpdate = true;
+                this.currentItemID = this.furnaceItemStacks[0].itemID;
+            }
+
+            if (this.currentItemID != 0 && this.furnaceItemStacks[0] == null)
+            {
+                shouldUpdate = true;
+                this.currentItemID = 0;
+            }
+
+            if (this.isBurning() && this.canSmelt() && (this.furnaceItemStacks[0].itemID == this.currentItemID))
@@ -286,0 +301 @@ public class TileEntityFurnace extends TileEntity implements ISidedInventory
+                this.currentItemID = this.furnaceItemStacks[0] == null ? 0 : this.furnaceItemStacks[0].itemID;
@@ -292 +307,6 @@ public class TileEntityFurnace extends TileEntity implements ISidedInventory
-                BlockFurnace.updateFurnaceBlockState(this.furnaceBurnTime > 0, this.worldObj, this.xCoord, this.yCoord, this.zCoord);
+                shouldUpdate = true;
+            }
+
+            if (shouldUpdate)
+            {
+                BlockFurnace.updateFurnaceBlockState(this.furnaceBurnTime > 0, this.worldObj, this.xCoord, this.yCoord, this.zCoord, this.furnaceItemStacks[0] != null || this.furnaceItemStacks[2] != null);
