diff --git a/src/minecraft_server/net/minecraft/src/AnvilChunkLoader.java b/src/minecraft_server/net/minecraft/src/AnvilChunkLoader.java
index 885046e..e832e60 100644
--- a/src/minecraft_server/net/minecraft/src/AnvilChunkLoader.java
+++ b/src/minecraft_server/net/minecraft/src/AnvilChunkLoader.java
@@ -84,0 +85,2 @@ public class AnvilChunkLoader implements IChunkLoader, IThreadedFileIO
+            GPEBTWTweak.onLoadChunk(par1World, var5);
+
@@ -106,0 +109 @@ public class AnvilChunkLoader implements IChunkLoader, IThreadedFileIO
+            GPEBTWTweak.onSaveChunk(par1World, par2Chunk);
