diff --git a/src/minecraft_server/net/minecraft/src/BlockLadder.java b/src/minecraft_server/net/minecraft/src/BlockLadder.java
index 19d6dd1..385411f 100644
--- a/src/minecraft_server/net/minecraft/src/BlockLadder.java
+++ b/src/minecraft_server/net/minecraft/src/BlockLadder.java
@@ -166,0 +167,15 @@ public class BlockLadder extends Block
+
+    public int GetItemIDDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return Item.stick.itemID;
+    }
+
+    public int GetItemCountDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return 2;
+    }
+
+    public int GetItemDamageDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return 0;
+    }
