diff --git a/src/minecraft_server/net/minecraft/src/BlockNote.java b/src/minecraft_server/net/minecraft/src/BlockNote.java
index 20e7f37..0f2430c 100644
--- a/src/minecraft_server/net/minecraft/src/BlockNote.java
+++ b/src/minecraft_server/net/minecraft/src/BlockNote.java
@@ -110,0 +111,12 @@ public class BlockNote extends BlockContainer
+
+    public boolean OnBlockSawed(World world, int x, int y, int z)
+    {
+        int i;
+        for (i = 0; i < 6; ++i)
+        {
+            FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, FCBetterThanWolves.fcBlockWoodSidingItemStubID, 0);
+        }
+        FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, FCBetterThanWolves.fcItemRedstoneLatch.itemID, 0);
+        world.setBlockToAir(x, y, z);
+        return true;
+    }
