diff --git a/src/minecraft_server/net/minecraft/src/BlockBookshelf.java b/src/minecraft_server/net/minecraft/src/BlockBookshelf.java
index cc269b8..6d07fe6 100644
--- a/src/minecraft_server/net/minecraft/src/BlockBookshelf.java
+++ b/src/minecraft_server/net/minecraft/src/BlockBookshelf.java
@@ -27,0 +28,15 @@ public class BlockBookshelf extends Block
+
+    public boolean OnBlockSawed(World world, int x, int y, int z)
+    {
+        int i;
+        for (i = 0; i < 4; ++i)
+        {
+            FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, FCBetterThanWolves.fcBlockWoodSidingItemStubID, 0);
+        }
+        for (i = 0; i < 3; ++i)
+        {
+            FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, Item.book.itemID, 0);
+        }
+        world.setBlockToAir(x, y, z);
+        return true;
+    }
