diff --git a/src/minecraft_server/net/minecraft/src/BlockTrapDoor.java b/src/minecraft_server/net/minecraft/src/BlockTrapDoor.java
index d6cbc28..0c425b4 100644
--- a/src/minecraft_server/net/minecraft/src/BlockTrapDoor.java
+++ b/src/minecraft_server/net/minecraft/src/BlockTrapDoor.java
@@ -294,0 +295,15 @@ public class BlockTrapDoor extends Block
+
+    public int GetItemIDDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return FCBetterThanWolves.fcBlockWoodSidingItemStubID;
+    }
+
+    public int GetItemCountDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return 2;
+    }
+
+    public int GetItemDamageDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return 0;
+    }
