diff --git a/src/minecraft_server/net/minecraft/src/BlockFenceGate.java b/src/minecraft_server/net/minecraft/src/BlockFenceGate.java
index 18c4956..22c9bae 100644
--- a/src/minecraft_server/net/minecraft/src/BlockFenceGate.java
+++ b/src/minecraft_server/net/minecraft/src/BlockFenceGate.java
@@ -146,0 +147,15 @@ public class BlockFenceGate extends BlockDirectional
+
+    public int GetItemIDDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return FCBetterThanWolves.fcBlockWoodMouldingItemStubID;
+    }
+
+    public int GetItemCountDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return 3;
+    }
+
+    public int GetItemDamageDroppedOnSaw(World var1, int var2, int var3, int var4)
+    {
+        return 0;
+    }
