diff --git a/src/minecraft_server/net/minecraft/src/EntityEnderman.java b/src/minecraft_server/net/minecraft/src/EntityEnderman.java
index 641bb4c..dcf09ee 100644
--- a/src/minecraft_server/net/minecraft/src/EntityEnderman.java
+++ b/src/minecraft_server/net/minecraft/src/EntityEnderman.java
@@ -712,0 +713,5 @@ public class EntityEnderman extends EntityMob
+    protected void dropHead()
+    {
+        this.entityDropItem(new ItemStack(Item.skull.itemID, 1, 6), 0.0F);
+    }
+
