diff --git a/src/minecraft_server/net/minecraft/src/EntityPigZombie.java b/src/minecraft_server/net/minecraft/src/EntityPigZombie.java
index dedfb12..8801b60 100644
--- a/src/minecraft_server/net/minecraft/src/EntityPigZombie.java
+++ b/src/minecraft_server/net/minecraft/src/EntityPigZombie.java
@@ -224 +224,4 @@ public class EntityPigZombie extends EntityZombie
-    protected void dropHead() {}
+    protected void dropHead()
+    {
+        this.entityDropItem(new ItemStack(Item.skull.itemID, 1, 7), 0.0F);
+    }
