diff --git a/src/minecraft_server/net/minecraft/src/EntitySpider.java b/src/minecraft_server/net/minecraft/src/EntitySpider.java
index f65922d..48eef55 100644
--- a/src/minecraft_server/net/minecraft/src/EntitySpider.java
+++ b/src/minecraft_server/net/minecraft/src/EntitySpider.java
@@ -304,0 +305,5 @@ public class EntitySpider extends EntityMob
+    protected void dropHead()
+    {
+        this.entityDropItem(new ItemStack(Item.skull.itemID, 1, 5), 0.0F);
+    }
+
