diff --git a/src/minecraft_server/net/minecraft/src/FCBlockWoodSidingAndCornerAndDecorative.java b/src/minecraft_server/net/minecraft/src/FCBlockWoodSidingAndCornerAndDecorative.java
index d67b732..1fbc5ed 100644
--- a/src/minecraft_server/net/minecraft/src/FCBlockWoodSidingAndCornerAndDecorative.java
+++ b/src/minecraft_server/net/minecraft/src/FCBlockWoodSidingAndCornerAndDecorative.java
@@ -66 +66 @@ public class FCBlockWoodSidingAndCornerAndDecorative extends FCBlockSidingAndCor
-        return var5 == 12 ? super.GetItemIDDroppedOnSaw(var1, var2, var3, var4) : (var5 == 14 ? FCBetterThanWolves.fcBlockWoodCornerItemStubID : (this.GetIsCorner(var1, var2, var3, var4) ? FCBetterThanWolves.fcGear.itemID : FCBetterThanWolves.fcBlockWoodMouldingItemStubID));
+        return var5 == 12 || var5 == 14 ? FCBetterThanWolves.fcBlockWoodCornerItemStubID : (this.GetIsCorner(var1, var2, var3, var4) ? FCBetterThanWolves.fcGear.itemID : FCBetterThanWolves.fcBlockWoodMouldingItemStubID);
@@ -74 +74 @@ public class FCBlockWoodSidingAndCornerAndDecorative extends FCBlockSidingAndCor
-            return var5 == 14 ? 2 : super.GetItemCountDroppedOnSaw(var1, var2, var3, var4);
+            return var5 == 12 || var5 == 14 ? 2 : super.GetItemCountDroppedOnSaw(var1, var2, var3, var4);
@@ -87 +87 @@ public class FCBlockWoodSidingAndCornerAndDecorative extends FCBlockSidingAndCor
-            return var5 == 14 ? this.GetWoodTypeFromBlockID() : super.GetItemDamageDroppedOnSaw(var1, var2, var3, var4);
+            return var5 == 12 || var5 == 14 ? this.GetWoodTypeFromBlockID() : super.GetItemDamageDroppedOnSaw(var1, var2, var3, var4);
