diff --git a/src/minecraft_server/net/minecraft/src/FCBlockGravel.java b/src/minecraft_server/net/minecraft/src/FCBlockGravel.java
index 340eedd..41dfc25 100644
--- a/src/minecraft_server/net/minecraft/src/FCBlockGravel.java
+++ b/src/minecraft_server/net/minecraft/src/FCBlockGravel.java
@@ -50,0 +51,8 @@ public class FCBlockGravel extends BlockGravel
+        if (var1.rand.nextInt(10) == 0)
+        {
+            if (var1.rand.nextFloat() <= var6)
+            {
+                this.dropBlockAsItem_do(var1, var2, var3, var4, new ItemStack(Item.flint));
+            }
+            return;
+        }
