diff --git a/src/minecraft_server/net/minecraft/src/FCBlockAxle.java b/src/minecraft_server/net/minecraft/src/FCBlockAxle.java
index 2c72324..21ac5db 100644
--- a/src/minecraft_server/net/minecraft/src/FCBlockAxle.java
+++ b/src/minecraft_server/net/minecraft/src/FCBlockAxle.java
@@ -522,0 +523,12 @@ public class FCBlockAxle extends Block implements FCIBlock
+
+    public boolean OnBlockSawed(World world, int x, int y, int z)
+    {
+        int i;
+        for (i = 0; i < 2; ++i)
+        {
+            FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, FCBetterThanWolves.fcBlockWoodCornerItemStubID, 0);
+        }
+        FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, FCBetterThanWolves.fcRopeItem.itemID, 0);
+        world.setBlockToAir(x, y, z);
+        return true;
+    }
