diff --git a/src/minecraft_server/net/minecraft/src/FCBlockBlockDispenser.java b/src/minecraft_server/net/minecraft/src/FCBlockBlockDispenser.java
index 5503f7f..55fb3c3 100644
--- a/src/minecraft_server/net/minecraft/src/FCBlockBlockDispenser.java
+++ b/src/minecraft_server/net/minecraft/src/FCBlockBlockDispenser.java
@@ -757,0 +758,10 @@ public class FCBlockBlockDispenser extends BlockContainer implements FCIBlock
+                                else if (var17.itemID == GPEBTWTweak.gpeItemLooseRock.itemID)
+                                {
+                                    var34 = FCUtilsInventory.CountItemsInInventory(var16, var17.itemID, 32767);
+                                    if (var34 >= 3)
+                                    {
+                                        FCUtilsInventory.ConsumeItemsInInventory(var16, var17.itemID, 32767, 3);
+                                        var33 = Block.cobblestone;
+                                    }
+                                    else var12 = true;
+                                }
