diff --git a/src/minecraft_server/net/minecraft/src/FCBlockHopper.java b/src/minecraft_server/net/minecraft/src/FCBlockHopper.java
index f26c9b0..851cd24 100644
--- a/src/minecraft_server/net/minecraft/src/FCBlockHopper.java
+++ b/src/minecraft_server/net/minecraft/src/FCBlockHopper.java
@@ -486,4 +486,16 @@ public class FCBlockHopper extends BlockContainer implements FCIBlockMechanical,
-                            ItemStack var17 = new ItemStack(Item.flint.itemID, var20, 0);
-                            EntityItem var18 = new EntityItem(var1, var5.posX, var5.posY, var5.posZ, var17);
-                            var18.delayBeforeCanPickup = 10;
-                            var1.spawnEntityInWorld(var18);
+                            int numflints = var20 - var1.rand.nextInt(10 * var20);
+                            if (numflints < 0) numflints = 0;
+                            if (var20 - numflints > 0)
+                            {
+                                ItemStack stack = new ItemStack(GPEBTWTweak.gpeItemLooseRock.itemID, var20 - numflints, 0);
+                                EntityItem rocks = new EntityItem(var1, var5.posX, var5.posY, var5.posZ, stack);
+                                rocks.delayBeforeCanPickup = 10;
+                                var1.spawnEntityInWorld(rocks);
+                            }
+                            if (numflints > 0)
+                            {
+                                ItemStack stack = new ItemStack(Item.flint.itemID, numflints, 0);
+                                EntityItem flints = new EntityItem(var1, var5.posX, var5.posY, var5.posZ, stack);
+                                flints.delayBeforeCanPickup = 10;
+                                var1.spawnEntityInWorld(flints);
+                            }
