diff --git a/src/minecraft_server/net/minecraft/src/FCBlockPumpkin.java b/src/minecraft_server/net/minecraft/src/FCBlockPumpkin.java
index 269babe..df49f1b 100644
--- a/src/minecraft_server/net/minecraft/src/FCBlockPumpkin.java
+++ b/src/minecraft_server/net/minecraft/src/FCBlockPumpkin.java
@@ -5,0 +6 @@ public class FCBlockPumpkin extends FCBlockGourd
+    private boolean isLantern;
@@ -9,0 +11 @@ public class FCBlockPumpkin extends FCBlockGourd
+        this.isLantern = var2;
@@ -112,0 +115,19 @@ public class FCBlockPumpkin extends FCBlockGourd
+    protected void Explode(World var1, double var2, double var4, double var6)
+    {
+        super.Explode(var1, var2, var4, var6);
+        if (this.isLantern)
+        {
+            int x = MathHelper.floor_double(var2);
+            int y = MathHelper.floor_double(var4);
+            int z = MathHelper.floor_double(var6);
+            if (Block.torchWood.canPlaceBlockAt(var1, x, y, z))
+            {
+                var1.setBlock(x, y, z, Block.torchWood.blockID);
+            }
+            else
+            {
+                FCUtilsItem.EjectStackWithRandomVelocity(var1, x, y, z, new ItemStack(Block.torchWood, 1, 0));
+            }
+        }
+    }
+
