diff --git a/src/minecraft_server/net/minecraft/src/FCBlockScrewPump.java b/src/minecraft_server/net/minecraft/src/FCBlockScrewPump.java
index 70b1347..404ff6d 100644
--- a/src/minecraft_server/net/minecraft/src/FCBlockScrewPump.java
+++ b/src/minecraft_server/net/minecraft/src/FCBlockScrewPump.java
@@ -604,0 +605,13 @@ public class FCBlockScrewPump extends Block implements FCIBlock, FCIBlockMechani
+
+    public boolean OnBlockSawed(World world, int x, int y, int z)
+    {
+        int i;
+        for (i = 0; i < 3; ++i)
+        {
+            FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, FCBetterThanWolves.fcBlockWoodSidingItemStubID, 0);
+        }
+        FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, FCBetterThanWolves.fcGrate.itemID, 0);
+        FCUtilsItem.EjectSingleItemWithRandomOffset(world, x, y, z, FCBetterThanWolves.fcItemScrew.itemID, 0);
+        world.setBlockToAir(x, y, z);
+        return true;
+    }
