diff --git a/src/minecraft_server/net/minecraft/src/FCContainerInfernalEnchanter.java b/src/minecraft_server/net/minecraft/src/FCContainerInfernalEnchanter.java
index 3013ac5..8bf738d 100644
--- a/src/minecraft_server/net/minecraft/src/FCContainerInfernalEnchanter.java
+++ b/src/minecraft_server/net/minecraft/src/FCContainerInfernalEnchanter.java
@@ -289 +289 @@ public class FCContainerInfernalEnchanter extends Container
-                    if (this.IsEnchantmentAppropriateForItem(var4, var2) && !this.DoesEnchantmentConflictWithExistingOnes(var4, var2))
+                    if (!this.IsEnchantmentValid(var4) || (this.IsEnchantmentAppropriateForItem(var4, var2) && !this.DoesEnchantmentConflictWithExistingOnes(var4, var2)))
@@ -290,0 +291 @@ public class FCContainerInfernalEnchanter extends Container
+                        if (!this.IsEnchantmentValid(var4)) var4 = Enchantment.unbreaking.effectId;
@@ -319,0 +321,5 @@ public class FCContainerInfernalEnchanter extends Container
+    private boolean IsEnchantmentValid(int i)
+    {
+        return i < Enchantment.enchantmentsList.length && Enchantment.enchantmentsList[i] != null;
+    }
+
@@ -322 +328 @@ public class FCContainerInfernalEnchanter extends Container
-        return Enchantment.enchantmentsList[var1].func_92089_a(var2);
+        return IsEnchantmentValid(var1) && Enchantment.enchantmentsList[var1].func_92089_a(var2);
@@ -381 +387,8 @@ public class FCContainerInfernalEnchanter extends Container
-                            var6.addEnchantment(Enchantment.enchantmentsList[var7], var2 + 1);
+                            if (this.IsEnchantmentValid(var7))
+                            {
+                                var6.addEnchantment(Enchantment.enchantmentsList[var7], var2 + 1);
+                            }
+                            else
+                            {
+                                this.m_localWorld.createExplosion((Entity)null, (double)this.m_iBlockI + 0.5D, (double)this.m_iBlockJ + 2.0D, (double)this.m_iBlockK + 0.5D, (float)var3 * 0.5F, true);
+                            }
